src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -39,12 +42,77 @@ import {
   defaultDataSourceSelection,
 } from '../service/data_source_selection_service';
 import { DataSourceError } from '../types';
+import { DATACONNECTIONS_BASE } from '../constants';
+
+export const getDirectQueryConnections = async (dataSourceId: string, http: HttpSetup) => {
+  const endpoint = `${DATACONNECTIONS_BASE}/dataSourceMDSId=${dataSourceId}`;
+  const res = await http.get(endpoint);
+  const directQueryConnections: DataSourceTableItem[] = res.map(
+    (dataConnection: DirectQueryDatasourceDetails) => ({
+      id: `${dataSourceId}-${dataConnection.name}`,
+      title: dataConnection.name,
+      type: dataConnection.connector === 'S3GLUE' ? 'Amazon S3' : 'Prometheus',
```

Comment:
This is not scalable. It would be better to check the `connector`, filter out incompatible ones, and then show the correct type.

Think of a scenario where a future version of OSD is used to configure connectors and then OSD 2.17 is used by some user. OSD 2.17 will assume those connectors are Prometheus connectors.